File: breakpad/app/src/main/java/org/devio/as/proj/breakpad/MainActivity.java
package org.devio.as.proj.breakpad;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import org.devio.as.proj.libbreakpad.NativeCrashHandler;

import java.io.File;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'native-lib' library on application startup.
    static {
        System.loadLibrary("native-lib");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Example of a call to a native method
        TextView tv = findViewById(R.id.sample_text);
        tv.setText(stringFromJNI());

        tv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                File crashDir = new File(getCacheDir(), "native_crash");
                if (!crashDir.exists()) {
                    crashDir.mkdirs();
                }
                NativeCrashHandler.init(crashDir.getAbsolutePath());
                crash();
            }
        });
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();

    public native void crash();
}

File: as-kotlin/app/src/main/java/org/devio/as/hi/kotlin_dmeo/MainActivity.kt
package org.devio.`as`.hi.kotlin_dmeo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        val textView = find<TextView>(R.id.test)
        R.id.test.onClick(this) {
            test.text = "Kotlin扩展"
        }

    }

    fun test(str: String?) {
        if (str.isNullOrEmpty()) {
            println("str.isNullOrEmpty():${str.isNullOrEmpty()}")
        } else if (str.isNullOrBlank()) {
            println("str.isNullOrBlank():${str.isNullOrBlank()}")
        }
    }
}
